Fix several reported issues in String's documentation

Mickeon opened the pull request:
Closes https://github.com/godotengine/godot-docs/issues/8043
Closes https://github.com/godotengine/godot-docs/issues/8044
Closes https://github.com/godotengine/godot/issues/86734
Mitigates https://github.com/godotengine/godot-docs/issues/8190

This PR updates the String's class reference slightly.
It formats uppercase and lowercase in their respective String methods and removes a seemingly useless note in `String.capitalize()` and incorporates it with the example code, instead.
~~Finally, for StringName, it adds a note exclusive to C#, because all of the listed methods are not available there. This note is almost word-for-word taken from the bug report https://github.com/godotengine/godot-docs/issues/8190 (@raulsntos)~~ 

Oh great heavens, news flash. Let's also add some examples for https://github.com/godotengine/godot/issues/86734, why not.

Could be easily cherrypicked for prior Godot 4.x versions

Mickeon committed:
Fixing some reports to String's documentation

Closes https://github.com/godotengine/godot-docs/issues/8043
Closes https://github.com/godotengine/godot-docs/issues/8044
Mitigates https://github.com/godotengine/godot-docs/issues/8190
Closes https://github.com/godotengine/godot/issues/86734

raulsntos commented on doc/classes/StringName.xml:
```diff
@@ -8,6 +8,7 @@
 		You will usually just pass a [String] to methods expecting a [StringName] and it will be automatically converted, but you may occasionally want to construct a [StringName] ahead of time with the [StringName] constructor or, in GDScript, the literal syntax [code]&amp;"example"[/code].
 		See also [NodePath], which is a similar concept specifically designed to store pre-parsed scene tree paths.
 		All of [String]'s methods are available in this class too. They convert the [StringName] into a string, and they also return a string. This is highly inefficient and should only be used if the string is desired.
+		[b]Note:[/b] In C#, none of the methods listed on this page are available. Instead, use the provided [code]ToString()[/code] method to cast a [StringName] to a string, and then use the equivalent methods in [code]System.String[/code] or [code]StringExtensions[/code]. For convenience, a list of equivalent methods is provided [url=$DOCS_URL/tutorials/scripting/c_sharp/c_sharp_differences.html#str ...
```
Just a nit pick, but I disagree a bit with the wording of the note about these methods not being available. AFAIK GDScript is just implicitly converting the StringName to String behind your back, so using `ToString` in C# is doing the same thing but more explicitly.

We discussed in [RC](https://chat.godotengine.org/channel/dotnet?msg=hKNuwHB89usJxrKy2) a while ago that StringName should be added to the list of classes with notable differences in C#:

https://github.com/godotengine/godot/blob/179dfdc8d78b5bd5377dd115af026df58308bdaf/doc/tools/make_rst.py#L105

It was an oversight when I added this note in https://github.com/godotengine/godot/pull/79206.

Mickeon replied:
Okiedokie. What are you suggesting the note should say then, if anything?

raulsntos replied:
How about something like this?

```suggestion
		[b]Note:[/b] In C#, an explicit conversion to [code]System.String[/code] is required to use the methods listed on this page. Use the [code]ToString()[/code] method to cast a [StringName] to a string, and then use the equivalent methods in [code]System.String[/code] or [code]StringExtensions[/code]. For convenience, a list of equivalent methods is provided [url=$DOCS_URL/tutorials/scripting/c_sharp/c_sharp_differences.html#string]here[/url].
```

We probably don't even need the reference to the differences documentation page since it would be in the note added when the `StringName` class is added to the `CLASSES_WITH_CSHARP_DIFFERENCES` list. For reference, the note looks like this:

![image](https://github.com/godotengine/godot/assets/3903059/310e8f34-9bd3-4345-8304-35751b741a16)

jsjtxietian commented on doc/classes/String.xml:
```diff
@@ -96,13 +96,14 @@
+				"2D, FPS, PNG".Capitalize();   // Returns "2d, Fps, Png"
 				[/csharp]
 				[/codeblocks]
-				[b]Note:[/b] This method not the same as the default appearance of properties in the Inspector dock, as it does not capitalize acronyms ([code]"2D"[/code], [code]"FPS"[/code], [code]"PNG"[/code], etc.) as you may expect.
```
This method "is" not the same as the default appearance of properties in the Inspector dock ?

Mickeon replied:
Red and the minus sign means that this PR removes the sentence outright. Yeah, it was missing the "is", too.

Mickeon commented:
Sounds good. I changed and removed part of the **StringName** C# note. I would also say that the PR no longer fully addresses https://github.com/godotengine/godot-docs/issues/8190 because it needs to be stated clearly in the C# differences page, too.

raulsntos commented:
I think you could make the change to `CLASSES_WITH_CSHARP_DIFFERENCES` in this PR:

```diff
diff --git a/doc/tools/make_rst.py b/doc/tools/make_rst.py
index c3a21f3d7b..df3e928538 100755
--- a/doc/tools/make_rst.py
+++ b/doc/tools/make_rst.py
@@ -105,6 +105,7 @@ EDITOR_CLASSES: List[str] = [
 CLASSES_WITH_CSHARP_DIFFERENCES: List[str] = [
     "@GlobalScope",
     "String",
+    "StringName",
     "NodePath",
     "Signal",
     "Callable",

```

Mickeon commented:
Oh that's neat, I did not realise part of it could have been done in the same repository. Done.

raulsntos reviewed (approved):
LGTM.

Mickeon commented:
Addressed the above feedback with an additional note. I didn't want to add the explanation in the same line, to keep the short summary consistent to the other similar methods.

akien-mga commented:
Thanks!

YuriSizov commented:
Cherry-picked for 4.2.2.

YuriSizov commented:
Cherry-picked for 4.1.4.
